rclpy/test/test_time.py

```diff
@@ -51,12 +51,11 @@ def test_duration_construction(self):
         assert duration.nanoseconds == 1500000000
 
         with self.assertRaises(OverflowError):
-            duration = Duration(nanoseconds=2**64)
-        duration = Duration(nanoseconds=2**64 - 1)
-        assert duration.nanoseconds == 2**64 - 1
+            duration = Duration(nanoseconds=2**63)
+        duration = Duration(nanoseconds=2**63 - 1)
+        assert duration.nanoseconds == 2**63 - 1
 
-        with self.assertRaises(ValueError):
-            duration = Duration(seconds=-1)
+        duration = Duration(seconds=-1)
```

Comment:
It would be good to assert the resulting `duration.nanoseconds` in this case. Maybe also try the lowest possible value?

Reply:
Added check on resulting `duration.nanoseconds` plus a check on lowest possible value 7697213d798ce99b50f13dddf6936fbc90be4d13

The float seconds to integer nanoseconds precision isn't very good at that range. On my machine `Duration(seconds=-2**63 / 1e9 - 954e-9)` is the first negative floating point number to `OverflowError`, but I'm not sure how portable that is so I did `-2**63 / 1e9 - 1` instead.